### ISSUE TITLE
[Marketplace] Receive error when browsing by Tag

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#2626]
### Expected behavior
No errors

### Actual Behavior
Receive error in console
```
modules.js?hash=6d2e3df…:84778 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the TagNavContainer component.
```

### Steps to Reproduce the Behavior
1. Create a new product
1. Create a new tag
1. Add tag to product
1. Click on tag in nav
1. Observe error

**Note** This takes a little bit of work to replicate. It doesn't happen every time but if you flip back and forth between tags and then wait a second or two on the added tag page you can see the error.

### Versions
```
Node: 7.10.0
NPM: 5.0.3
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.5.0
Reaction branch: marketplace
Docker: 17.06.0-ce
commit: b6e18a4b5a35b852bd75648ae395a2f4b77eae81
```

**Comment by mikemurray** [comment]
Can reproduce consistently by having a submenu under one of the tags. While on the PDP page clicking on that tag (which is a dropdown ), switching page, then hovering over that tag again produces the error.

I usually see this when a component tries to call `this.setState()` in a lifecycle method like `componentDidMount()` / `componentWillReceiveProps` or in delayed or debounded set states.

Also, note that the main navbar may be rendering between page views, so the navbar you're clicking on was probably unmounted and a new instance re-attached. Thus, any `setState` calls that were delayed from the old instance would cause the error to show.

**Comment by zenweasel** [comment]
Closed via #2796
